### PR TITLE
Add attachment from payload

### PR DESCRIPTION
**Pull request by Shakila**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java [hunk 1]
```diff
@@ -295,6 +312,49 @@ private void addAttachment(MimeMultipart multipart, String filePath) throws Mess
...
+                multipart.addBodyPart(part);
+            }
+        } catch (JSONException e) {
+            throw new IOException("Invalid JSON data format. " + e.getMessage());
+        }
+    }
+
+    private boolean isBase64(String stringBase64){
+        String regex = "([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{4}|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)";
+        Pattern pattern = Pattern.compile(regex);
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java]: Shall we have the Pattern as a static class variable?

File: src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java [hunk 2]
```diff
@@ -295,6 +312,49 @@ private void addAttachment(MimeMultipart multipart, String filePath) throws Mess
...
+        if (!pattern.matcher(stringBase64).matches()) {
+            return false;
+        } else {
+            return true;
+        }
```
Comment (Nirothipan) [inline src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java]: shall we simplify this to return pattern.matcher(stringBase64).matches();

File: src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java [hunk 3]
```diff
@@ -295,6 +312,49 @@ private void addAttachment(MimeMultipart multipart, String filePath) throws Mess
...
+                String fileContentType = tika.detect(bain);
+                DataHandler dataHandler = new DataHandler(new EmailAttachmentDataSource(attachmentName, bain,
+                        fileContentType));
+                part.setDataHandler(dataHandler);
+                part.setHeader(CONTENT_TYPE_HEADER, dataHandler.getContentType());
+                part.setHeader(CONTENT_TRANSFER_ENCODING_HEADER, this.contentTransferEncoding);
+                multipart.addBodyPart(part);
+            }
+        } catch (JSONException e) {
+            throw new IOException("Invalid JSON data format. " + e.getMessage());
```
Comment (Nirothipan) [inline src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java]: do we not need the object e here.

File: src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java [hunk 4]
```diff
@@ -242,9 +248,20 @@ public MimeMessage build() throws MessagingException, IOException {
             MimeBodyPart body = new MimeBodyPart();
             setMessageContent(body);
             multipart.addBodyPart(body);
-            String[] attachFiles = attachments.split(",");
-            for (String filePath : attachFiles) {
-                addAttachment(multipart, filePath);
+            if (attachments.startsWith("[")) {
```
Comment (Nirothipan) [inline src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java:254]: shall we add a comment with the reason for this check

Comment (Shakila) [reply src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java:254]: Done

File: src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java (hunk 2 above)
Comment (Shakila) [reply src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java]: Done

File: src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java (hunk 1 above)
Comment (Shakila) [reply src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java]: Done

File: src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java (hunk 3 above)
Comment (Shakila) [reply src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java]: Done

File: src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java (hunk 1 above)
Comment (SanojPunchihewa) [reply src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java]: Shall we have the compiled pattern also as a static class variable? Compiling it every time may cause performance impacts.

File: src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java [hunk 5]
```diff
@@ -295,6 +314,44 @@ private void addAttachment(MimeMultipart multipart, String filePath) throws Mess
...
+                part.setHeader(CONTENT_TRANSFER_ENCODING_HEADER, this.contentTransferEncoding);
+                multipart.addBodyPart(part);
+            }
+        } catch (JSONException e) {
+            throw new IOException("Invalid JSON data format.", e);
+        }
+    }
+
+    private boolean isBase64(String stringBase64){
+        Pattern pattern = Pattern.compile(REGEX);
```
Comment (Nirothipan) [inline src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java]: we can do this also in class level?

Comment (Shakila) [reply src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java]: Done

File: src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java [hunk 6]
```diff
@@ -295,6 +315,43 @@ private void addAttachment(MimeMultipart multipart, String filePath) throws Mess
...
+     * Get content from request body and add the attachment
+     *
+     * @param multipart Multi part body the messages should be added to
+     * @param attachment  the attachment
+     * @throws MessagingException if failed to set attachments
+     * @throws IOException        if an error occurred while reading attachment content
+     */
+    private void addAttachment(MimeMultipart multipart, JSONObject attachment) throws MessagingException, IOException {
+
+        String attachmentName;
```
Comment (Nirothipan) [inline src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java]: shall we move this definition also inside try

**Review by Nirothipan — APPROVED** [review]

File: src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java (hunk 6 above)
Comment (Shakila) [reply src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java]: Done

File: src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java (hunk 1 above)
Comment (Shakila) [reply src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java]: Done

**Review by SanojPunchihewa — APPROVED** [review]

**Pull request merged**